tests/api/api/test_secrets.py

```diff
@@ -39,3 +40,72 @@ def test_vault_create_project_secrets(db: Session, client: TestClient):
     )
     secrets = response.json()["secrets"]
     assert secrets == data["secrets"]
+
+
+# Set a namespace here to run the test_k8s_project_secrets test.
+k8s_namespace = ""
+
+
+# Need to be a fixture since the k8s_helper is initialized as part of te TestClient, so need
+# to perform this configuration prior to that, to make k8s_helper use proper namespace.
+@pytest.fixture(autouse=True)
+def set_k8s_namespace():
+    mlconf.namespace = k8s_namespace
+
+
+@pytest.mark.skipif(k8s_namespace == "", reason="no k8s namespace defined")
```

Comment:
I don't see a reason to make integration tests here (that won't run automatically).
either unit (and mock the k8s helper) or system (or best - both)

Reply:
There is not much to do unit-test on if mocking the k8s helper. 
I do find these tests useful when running locally on my local minikube, since they do test the functionality without having to setup a full system test to do so. Indeed these are skipped by default, but they are easily activated and IMO should remain as they bring a lot of value with minimal cost.

Reply:
Moved the test to system tests.